src/Hamiltonians/Momentum.jl

```diff
@@ -29,31 +29,44 @@ Base.show(io::IO, mom::Momentum{C}) where {C} = print(io, "Momentum($C; fold=$(m
 LOStructure(::Type{<:Momentum}) = IsDiagonal()
 num_offdiagonals(ham::Momentum, add) = 0
 
-@inline function _momentum(add::SingleComponentFockAddress, fold)
+function fold_mod(m, M)
+    m = mod(m, M)
+    return m > M ÷ 2 ? m - M : m
+end
+
+function momentum(add::SingleComponentFockAddress; fold=false)
     M = num_modes(add)
-    momentum = float(dot(-cld(M,2)+1:fld(M,2), OccupiedModeMap(add)))
+    res = 0
+    for i in OccupiedModeMap(add)
+        res += (i.mode - cld(M, 2)) * i.occnum
+    end
     if fold
-        return mod1(momentum + cld(M,2), M) - cld(M,2)
+        return float(fold_mod(res, M))
```

Comment:
A minor point: I would expect `momentum()` to return an integer since that is how we are representing discrete momentum. Conversion to `float` can happen when it is called by `diagonal_element`?

Reply:
Good idea! Implemented and tested.